spectral_cube/spectral_cube.py

```diff
@@ -2168,26 +2168,138 @@ def find_lines(self, velocity_offset=None, velocity_convention=None,
         return result
 
 
-class StokesSpectralCube(SpectralCube):
+VALID_STOKES = ['I', 'Q', 'U', 'V', 'RR', 'LL', 'RL', 'LR']
+
+class StokesSpectralCube(object):
 
     """
-    A class to store a spectral cube with multiple Stokes parameters. By
-    default, this will act like a Stokes I spectral cube, but other stokes
-    parameters can be accessed with attribute notation.
+    A class to store a spectral cube with multiple Stokes parameters.
+
+    The individual Stokes cubes will share masks.
     """
 
-    def __init__(self, data, wcs, mask=None, meta=None, header=None):
+    def __init__(self, stokes_data, mask=None, meta=None, fill_value=None):
+
+        self._stokes_data = stokes_data
+        self._meta = meta or {}
+        self._fill_value = fill_value
+
+        reference = tuple(stokes_data.keys())[0]
+
+        for component in stokes_data:
+
+            if not isinstance(stokes_data[component], SpectralCube):
+                raise TypeError("stokes_data should be a dictionary of SpectralCube objects")
+
+            if not wcs_utils.check_equality(stokes_data[component].wcs,
+                                            stokes_data[reference].wcs):
+                raise ValueError("All spectral cubes in stokes_data should have the same WCS")
+
+            if component not in VALID_STOKES:
+                raise ValueError("Invalid Stokes component: {0} - should be one of I, Q, U, V, RR, LL, RL, LR".format(component))
+                
+            if stokes_data[component].shape != stokes_data[reference].shape:
+                raise ValueError("All spectral cubes shoul have the same shape")
+                
+        self._wcs = stokes_data[reference].wcs
+        self._shape = stokes_data[reference].shape
+        
+        if isinstance(mask, BooleanArrayMask):
+            if mask.shape != self._shape:
+                raise ValueError("Mask shape is not broadcastable to data shape: {0} vs {1}".format(mask.shape, self._shape))
+
+        self._mask = mask
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @property
+    def mask(self):
+        """
+        The underlying mask
+        """
+        return self._mask
+
+    @property
+    def wcs(self):
+        return self._wcs
+
+    def __dir__(self):
+        return list(self._stokes_data.keys())
```

Comment:
Good catch - I used ``object.__dir__(self)`` instead of ``list(self.__dict__.keys())``